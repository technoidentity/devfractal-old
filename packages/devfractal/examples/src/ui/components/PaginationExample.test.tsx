import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import { PaginationExample } from './PaginationExample'

test('Pagination Example', () => {
  const component: ReactTestRenderer = renderer.create(<PaginationExample />)
  expect(component.toJSON()).toMatchSnapshot()
})
