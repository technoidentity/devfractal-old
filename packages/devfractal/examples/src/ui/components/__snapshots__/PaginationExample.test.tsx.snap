// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Example 1`] = `
<section
  className="section"
>
  <h1
    className="title"
  >
    Dynamic Pagination(CurrentPage:
    1
    )
  </h1>
  <section
    className="section"
  >
    <div
      className="field is-grouped is-grouped-right"
    >
      <div
        className="control"
      >
        <button
          className="button is-white"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <span
            className="icon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-step-backward fa-w-14 "
              data-icon="step-backward"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 468V44c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v176.4l195.5-181C352.1 22.3 384 36.6 384 64v384c0 27.4-31.9 41.7-52.5 24.6L136 292.7V468c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="control"
      >
        <button
          className="button is-white"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <span
            className="icon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-angle-double-left fa-w-14 "
              data-icon="angle-double-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="control"
      >
        <button
          className="button is-white"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="icon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-angle-double-right fa-w-14 "
              data-icon="angle-double-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="control"
      >
        <button
          className="button is-white"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="icon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-step-forward fa-w-14 "
              data-icon="step-forward"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M384 44v424c0 6.6-5.4 12-12 12h-48c-6.6 0-12-5.4-12-12V291.6l-195.5 181C95.9 489.7 64 475.4 64 448V64c0-27.4 31.9-41.7 52.5-24.6L312 219.3V44c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </section>
  <h1
    className="title"
  >
    Pagination example
  </h1>
  <section
    className="section"
  >
    <nav
      aria-label="pagination"
      className="pagination"
      role="navigation"
    >
      <a
        className="pagination-previous"
      >
        Previous
      </a>
      <a
        className="pagination-next"
      >
        Next page
      </a>
      <ul
        className="pagination-list"
      >
        <li>
          <a
            aria-label="Goto page 1"
            className="pagination-link is-current"
          >
            1
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 45"
            className="pagination-link"
          >
            45
          </a>
        </li>
        <li>
          <a
            aria-label="Page 46"
            className="pagination-link is-current"
          >
            46
          </a>
        </li>
        <li>
          <a
            aria-label="Goto page 47"
            className="pagination-link"
          >
            47
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 86"
            className="pagination-link"
          >
            86
          </a>
        </li>
      </ul>
    </nav>
  </section>
  <h1
    className="title"
  >
    Alignment
  </h1>
  <section
    className="section"
  >
    <div
      className="has-text-weight-bold"
    >
      Centered
    </div>
    <nav
      aria-label="pagination"
      className="pagination is-centered"
      role="navigation"
    >
      <a
        className="pagination-previous"
      >
        Previous
      </a>
      <a
        className="pagination-next"
      >
        Next page
      </a>
      <ul
        className="pagination-list"
      >
        <li>
          <a
            aria-label="Goto page 1"
            className="pagination-link is-current"
          >
            1
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 45"
            className="pagination-link"
          >
            45
          </a>
        </li>
        <li>
          <a
            aria-label="Page 46"
            className="pagination-link is-current"
          >
            46
          </a>
        </li>
        <li>
          <a
            aria-label="Goto page 47"
            className="pagination-link"
          >
            47
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 86"
            className="pagination-link"
          >
            86
          </a>
        </li>
      </ul>
    </nav>
  </section>
  <section
    className="section"
  >
    <div
      className="has-text-weight-bold"
    >
      Right 
    </div>
    <nav
      aria-label="pagination"
      className="pagination is-right"
      role="navigation"
    >
      <a
        className="pagination-previous"
      >
        Previous
      </a>
      <a
        className="pagination-next"
      >
        Next page
      </a>
      <ul
        className="pagination-list"
      >
        <li>
          <a
            aria-label="Goto page 1"
            className="pagination-link is-current"
          >
            1
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 45"
            className="pagination-link"
          >
            45
          </a>
        </li>
        <li>
          <a
            aria-label="Page 46"
            className="pagination-link is-current"
          >
            46
          </a>
        </li>
        <li>
          <a
            aria-label="Goto page 47"
            className="pagination-link"
          >
            47
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 86"
            className="pagination-link"
          >
            86
          </a>
        </li>
      </ul>
    </nav>
  </section>
  <h1
    className="title"
  >
    Styles
  </h1>
  <section
    className="section"
  >
    <nav
      aria-label="pagination"
      className="pagination is-rounded"
      role="navigation"
    >
      <a
        className="pagination-previous"
      >
        Previous
      </a>
      <a
        className="pagination-next"
      >
        Next page
      </a>
      <ul
        className="pagination-list"
      >
        <li>
          <a
            aria-label="Goto page 1"
            className="pagination-link is-current"
          >
            1
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 45"
            className="pagination-link"
          >
            45
          </a>
        </li>
        <li>
          <a
            aria-label="Page 46"
            className="pagination-link is-current"
          >
            46
          </a>
        </li>
        <li>
          <a
            aria-label="Goto page 47"
            className="pagination-link"
          >
            47
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 86"
            className="pagination-link"
          >
            86
          </a>
        </li>
      </ul>
    </nav>
  </section>
  <h1
    className="title"
  >
    Sizes
  </h1>
  <section
    className="section"
  >
    <div
      className="has-text-weight-bold"
    >
      normal
    </div>
    <nav
      aria-label="pagination"
      className="pagination"
      role="navigation"
    >
      <a
        className="pagination-previous"
      >
        Previous
      </a>
      <a
        className="pagination-next"
      >
        Next page
      </a>
      <ul
        className="pagination-list"
      >
        <li>
          <a
            aria-label="Goto page 1"
            className="pagination-link is-current"
          >
            1
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 45"
            className="pagination-link"
          >
            45
          </a>
        </li>
        <li>
          <a
            aria-label="Page 46"
            className="pagination-link is-current"
          >
            46
          </a>
        </li>
        <li>
          <a
            aria-label="Goto page 47"
            className="pagination-link"
          >
            47
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 86"
            className="pagination-link"
          >
            86
          </a>
        </li>
      </ul>
    </nav>
  </section>
  <section
    className="section"
  >
    <div
      className="has-text-weight-bold"
    >
      small
    </div>
    <nav
      aria-label="pagination"
      className="pagination is-small"
      role="navigation"
    >
      <a
        className="pagination-previous"
      >
        Previous
      </a>
      <a
        className="pagination-next"
      >
        Next page
      </a>
      <ul
        className="pagination-list"
      >
        <li>
          <a
            aria-label="Goto page 1"
            className="pagination-link is-current"
          >
            1
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 45"
            className="pagination-link"
          >
            45
          </a>
        </li>
        <li>
          <a
            aria-label="Page 46"
            className="pagination-link is-current"
          >
            46
          </a>
        </li>
        <li>
          <a
            aria-label="Goto page 47"
            className="pagination-link"
          >
            47
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 86"
            className="pagination-link"
          >
            86
          </a>
        </li>
      </ul>
    </nav>
  </section>
  <section
    className="section"
  >
    <div
      className="has-text-weight-bold"
    >
      medium
    </div>
    <nav
      aria-label="pagination"
      className="pagination is-medium"
      role="navigation"
    >
      <a
        className="pagination-previous"
      >
        Previous
      </a>
      <a
        className="pagination-next"
      >
        Next page
      </a>
      <ul
        className="pagination-list"
      >
        <li>
          <a
            aria-label="Goto page 1"
            className="pagination-link is-current"
          >
            1
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 45"
            className="pagination-link"
          >
            45
          </a>
        </li>
        <li>
          <a
            aria-label="Page 46"
            className="pagination-link is-current"
          >
            46
          </a>
        </li>
        <li>
          <a
            aria-label="Goto page 47"
            className="pagination-link"
          >
            47
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 86"
            className="pagination-link"
          >
            86
          </a>
        </li>
      </ul>
    </nav>
  </section>
  <section
    className="section"
  >
    <div
      className="has-text-weight-bold"
    >
      large
    </div>
    <nav
      aria-label="pagination"
      className="pagination is-large"
      role="navigation"
    >
      <a
        className="pagination-previous"
      >
        Previous
      </a>
      <a
        className="pagination-next"
      >
        Next page
      </a>
      <ul
        className="pagination-list"
      >
        <li>
          <a
            aria-label="Goto page 1"
            className="pagination-link is-current"
          >
            1
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 45"
            className="pagination-link"
          >
            45
          </a>
        </li>
        <li>
          <a
            aria-label="Page 46"
            className="pagination-link is-current"
          >
            46
          </a>
        </li>
        <li>
          <a
            aria-label="Goto page 47"
            className="pagination-link"
          >
            47
          </a>
        </li>
        <li>
          <span
            className="pagination-ellipsis"
          >
            …
          </span>
        </li>
        <li>
          <a
            aria-label="Goto page 86"
            className="pagination-link"
          >
            86
          </a>
        </li>
      </ul>
    </nav>
  </section>
</section>
`;
